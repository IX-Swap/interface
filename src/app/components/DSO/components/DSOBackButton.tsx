import React from 'react'
import { AppRouterLinkComponent } from 'components/AppRouterLink'
import { Button } from '@material-ui/core'
import { IssuanceRoute } from 'app/pages/issuance/router/config'
import { useParams } from 'react-router-dom'

export const DSOBackButton = () => {
  const params = useParams()

  return (
    <Button
      component={AppRouterLinkComponent}
      to={IssuanceRoute.list} // TODO: add logic for the create DSO screen
      params={params}
    >
      Cancel
    </Button>
  )
}
