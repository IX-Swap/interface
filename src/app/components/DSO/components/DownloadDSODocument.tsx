import React from 'react'
import { useDownloadRawFile } from 'hooks/useDownloadRawFile'
import { IconButton, Tooltip } from '@mui/material'
import { convertBlobToFile, openFileInNewTab } from 'hooks/utils'
import { Launch } from '@mui/icons-material'

export interface DownloadDSODocumentProps {
  dsoId: string
  documentId: string
  type?: 'document' | 'subscriptionDocument'
}

export const DownloadDSODocument = (props: DownloadDSODocumentProps) => {
  const { dsoId, documentId, type = 'document' } = props
  const uri =
    type === 'document'
      ? `/issuance/dso/dataroom/documents/raw/${dsoId}/${documentId}`
      : `/issuance/dso/dataroom/subscription/raw/${dsoId}`
  const [download, { isLoading }] = useDownloadRawFile(uri, {
    onSuccess: ({ data }) => {
      const file = convertBlobToFile(data, '')
      openFileInNewTab(file)
    }
  })
  const handleDownload = async () => await download()

  return (
    <Tooltip title='Download File'>
      <IconButton disabled={isLoading} onClick={handleDownload} size='large'>
        <Launch color='disabled' style={{ width: 23, height: 23 }} />
      </IconButton>
    </Tooltip>
  )
}
