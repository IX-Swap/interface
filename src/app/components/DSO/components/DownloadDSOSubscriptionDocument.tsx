import React from 'react'
import { useDownloadRawFile } from 'hooks/useDownloadRawFile'
import { Button, ButtonProps } from '@mui/material'
import { convertBlobToFile, openFileInNewTab } from 'hooks/utils'

export interface DownloadDSOSubscriptionDocumentProps extends ButtonProps {
  dsoId: string
}

export const DownloadDSOSubscriptionDocument = (
  props: DownloadDSOSubscriptionDocumentProps
) => {
  const { dsoId, children = 'Download', ...rest } = props
  const uri = `/issuance/dso/dataroom/subscription/raw/${dsoId}`
  const [download, { isLoading }] = useDownloadRawFile(uri, {
    onSuccess: ({ data }) => {
      const file = convertBlobToFile(data, '')
      openFileInNewTab(file)
    }
  })
  const handleDownload = async () => await download()

  return (
    <Button {...rest} onClick={handleDownload} disabled={isLoading}>
      {children}
    </Button>
  )
}
