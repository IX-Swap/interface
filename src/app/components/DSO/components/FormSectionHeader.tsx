import React from 'react'
import { Box, Typography } from '@material-ui/core'
import { useStyles } from 'app/components/DSO/components/FormSectionHeader.styles'
import { Variant } from '@material-ui/core/styles/createTypography'

export interface FormSectionHeaderProps {
  title: string
  variant?: Variant
}

export const FormSectionHeader = ({
  title,
  variant = 'h3'
}: FormSectionHeaderProps) => {
  const { formHeader } = useStyles()

  return (
    <Box className={formHeader} pb={0.5} mb={3}>
      <Typography variant={variant}>{title}</Typography>
    </Box>
  )
}
