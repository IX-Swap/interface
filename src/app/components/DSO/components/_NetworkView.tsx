import React from 'react'
import { useAllNetworks } from 'app/pages/accounts/pages/withdrawalAddresses/hooks/useAllNetworks'

export interface NetworkViewProps {
  networkId: string
}

export const NetworkView = (props: NetworkViewProps) => {
  const { networkId } = props
  const { data, isLoading } = useAllNetworks()
  const network = data?.find(network => network._id === networkId)

  return isLoading ? <>'...' </> : <>{network?.name}</>
}
