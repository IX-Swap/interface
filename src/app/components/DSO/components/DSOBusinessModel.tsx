import React from 'react'
import { TypedField } from 'components/form/TypedField'
import { RichTextEditor } from 'components/form/RichTextEditor'
import { wysiwygValueExtractor } from 'helpers/forms'
import { DSOContainer } from 'app/components/DSO/components/DSOContainer'
import { useFormContext } from 'react-hook-form'
import { DSOFormValues } from 'types/dso'

export const DSOBusinessModel = () => {
  const { control } = useFormContext<DSOFormValues>()

  return (
    <DSOContainer title='Business Model' item xs={12}>
      {/* @ts-ignore */}
      <TypedField
        control={control}
        component={RichTextEditor}
        customRenderer
        label='Business Model'
        name='businessModel'
        valueExtractor={wysiwygValueExtractor}
      />
    </DSOContainer>
  )
}
