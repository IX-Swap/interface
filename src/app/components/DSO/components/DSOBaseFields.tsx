import React from 'react'
import { Grid, TextField } from '@material-ui/core'
import { TypedField } from 'components/form/TypedField'
import { dateTimeValueExtractor, numericValueExtractor } from 'helpers/forms'
import { CorporateSelect } from 'components/form/CorporateSelect'
import { NetworkSelect } from 'components/form/NetworkSelect'
import { AssetSelect } from 'components/form/AssetSelect'
import { useFormContext } from 'react-hook-form'
import { DSOFormValues } from 'types/dso'
import { documentValueExtractor } from 'app/components/DSO/utils'
import { DataroomFileType } from 'config/dataroom'
import { DateTimePicker } from 'components/form/_DateTimePicker'
import { VSpacer } from 'components/VSpacer'
import { Dropzone } from 'components/dataroom/Dropzone'
import { CapitalStructureSelect } from 'components/form/CapitalStructureSelect'
import { NumericInput } from 'components/form/NumericInput'
import { positiveNumberFormat } from 'config/numberFormat'
import { FormSectionHeader } from 'app/components/DSO/components/FormSectionHeader'

export interface DSOBaseFieldsProps {
  isNew: boolean
  isLive: boolean
}

export const DSOBaseFields = (props: DSOBaseFieldsProps) => {
  const { isNew, isLive } = props
  const { control, watch } = useFormContext<DSOFormValues>()
  const capitalStructure = watch('capitalStructure')

  return (
    <Grid item>
      <VSpacer size='medium' />
      <Grid container direction='column' spacing={2}>
        <Grid item>
          <FormSectionHeader title='DSO Information' />
        </Grid>
        <Grid item>
          {/* @ts-ignore */}
          <TypedField
            customRenderer
            component={Dropzone}
            name='logo'
            label='Upload Logo'
            control={control}
            valueExtractor={documentValueExtractor}
            accept={DataroomFileType.image}
            documentInfo={{
              type: 'DSO Logo'
            }}
          />
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                control={control}
                component={CapitalStructureSelect}
                label='Capital Structure'
                name='capitalStructure'
                helperText='Offering terms will be changed based on your capital structure'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              <TypedField
                component={NetworkSelect}
                label='Network'
                name='network'
                disabled={!isNew}
                control={control}
                helperText='Select your blockchain network from the list'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              <TypedField
                component={TextField}
                label='Token Name'
                name='tokenName'
                disabled={isLive}
                control={control}
                helperText='Name of the token that describes your offering the best'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={2}>
              <TypedField
                component={TextField}
                label='Symbol'
                name='tokenSymbol'
                disabled={isLive}
                control={control}
                helperText='Token symbol'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                component={NumericInput}
                numberFormat={positiveNumberFormat}
                label='Decimal Places'
                name='decimalPlaces'
                disabled={capitalStructure?.toLowerCase() === 'equity'}
                control={control}
                helperText='Decimal Places'
                variant='outlined'
                valueExtractor={numericValueExtractor}
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={8}>
              <TypedField
                component={CorporateSelect}
                label='Corporate Structure'
                name='corporate'
                control={control}
                helperText='Select your corporate from the list'
                variant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={4}>
              <TypedField
                assetType='Currency'
                component={AssetSelect}
                label='Currency'
                name='currency'
                control={control}
                helperText='Your preferred currency'
                variant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
        <Grid item>
          <Grid container spacing={3}>
            <Grid item xs={12} md={6}>
              {/* @ts-ignore */}
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Launch Date'
                name='launchDate'
                control={control}
                disabled={isLive}
                valueExtractor={dateTimeValueExtractor}
                // @ts-expect-error
                defaultValue={null}
                helperText='Offering launch date'
                inputVariant='outlined'
              />
            </Grid>
            <Grid item xs={12} md={6}>
              {/* @ts-ignore */}
              <TypedField
                component={DateTimePicker}
                customRenderer
                label='Completion Date'
                name='completionDate'
                control={control}
                valueExtractor={dateTimeValueExtractor}
                // @ts-expect-error
                defaultValue={null}
                helperText='Offering completion date'
                inputVariant='outlined'
              />
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}
