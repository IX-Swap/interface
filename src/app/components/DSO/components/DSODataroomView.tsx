import React from 'react'
import { Grid, Typography } from '@mui/material'
import { DigitalSecurityOffering } from 'types/dso'
import { Documents } from 'app/pages/identity/components/CorporateIdentityView/Documents'
import { FormSectionHeader } from 'ui/FormSectionHeader/FormSectionHeader'
import { FieldContainer } from 'ui/FieldContainer/FieldContainer'

export interface DSODataroomViewProps {
  dso: DigitalSecurityOffering
}

export const DSODataroomView = ({ dso }: DSODataroomViewProps) => {
  const supportingDocuments = dso.documents.filter(
    doc => Object.values(doc).length > 0 && doc.type === 'Supporting Document'
  )

  const otherDocuments = dso.documents.filter(
    doc => Object.values(doc).length > 0 && doc.type === 'Other'
  )

  return (
    <Grid container direction='column' spacing={3}>
      <FieldContainer>
        <Grid item container direction={'column'} spacing={5}>
          <Grid item>
            <FormSectionHeader title='Documents' />
          </Grid>
          {dso.subscriptionDocument !== undefined && (
            <Grid item xs={12} container spacing={3}>
              <Grid item xs={12}>
                <Typography>Subscription Document</Typography>
              </Grid>
              <Grid item xs={12} container spacing={1}>
                <Grid item xs={12}>
                  <Documents documents={[dso.subscriptionDocument]} />
                </Grid>
              </Grid>
            </Grid>
          )}

          {supportingDocuments.length > 0 && (
            <Grid item xs={12} container spacing={2}>
              <Grid item xs={12}>
                <Typography>Supporting Documents</Typography>
              </Grid>
              <Grid item xs={12} container spacing={1}>
                <Grid item xs={12}>
                  <Documents documents={supportingDocuments} />
                </Grid>
              </Grid>
            </Grid>
          )}

          {otherDocuments.length > 0 && (
            <Grid item xs={12} container spacing={2}>
              <Grid item xs={12}>
                <Typography>Dataroom</Typography>
              </Grid>
              <Grid item xs={12} container spacing={1}>
                <Grid item xs={12}>
                  <Documents documents={otherDocuments} />
                </Grid>
              </Grid>
            </Grid>
          )}
        </Grid>
      </FieldContainer>
    </Grid>
  )
}
