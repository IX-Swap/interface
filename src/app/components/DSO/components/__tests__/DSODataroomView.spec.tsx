import React from 'react'
import { render } from 'test-utils'
import { DSODataroomView } from 'app/components/DSO/components/DSODataroomView'
import { dso } from '__fixtures__/authorizer'
import * as Typography from '@material-ui/core'
import { FormSectionHeader } from 'app/components/DSO/components/FormSectionHeader'
import { DataroomHeader } from 'components/dataroom/DataroomHeader'
import { DataroomViewRow } from 'components/dataroom/DataroomViewRow'

jest.mock('@material-ui/core/Typography', () => jest.fn(() => null))

jest.mock('app/components/DSO/components/FormSectionHeader', () => ({
  FormSectionHeader: jest.fn(() => null)
}))

jest.mock('components/dataroom/DataroomHeader', () => ({
  DataroomHeader: jest.fn(() => null)
}))

jest.mock('components/dataroom/DataroomViewRow', () => ({
  DataroomViewRow: jest.fn(() => null)
}))

describe('DSODataroomView', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it.skip('renders without error', () => {
    render(<DSODataroomView dso={dso} />)
  })

  it('renders title when showTitle is false', () => {
    render(<DSODataroomView dso={dso} showTitle={false} />)

    expect(FormSectionHeader).toHaveBeenCalledTimes(0)
  })

  it('renders title when showTitle is true', () => {
    render(<DSODataroomView dso={dso} showTitle />)

    expect(FormSectionHeader).toHaveBeenCalledTimes(1)
  })

  it('renders subscription document title with correct props', () => {
    render(<DSODataroomView dso={dso} />)

    expect(Typography).toHaveBeenNthCalledWith(
      1,
      expect.objectContaining({
        variant: 'h5',
        children: 'Subscription Document'
      }),
      {}
    )
  })

  it('renders subscription document title with correct props when showTitle is false', () => {
    render(<DSODataroomView dso={dso} />)

    expect(Typography).toHaveBeenNthCalledWith(
      1,
      expect.objectContaining({
        variant: 'h5',
        children: 'Subscription Document'
      }),
      {}
    )
  })

  it('renders message when subscription document is undefined ', () => {
    render(
      <DSODataroomView dso={{ ...dso, subscriptionDocument: undefined }} />
    )

    expect(Typography).toHaveBeenNthCalledWith(
      2,
      expect.objectContaining({
        children: 'No subscription document provided'
      }),
      {}
    )
  })

  it('renders subscription document components with correct props', () => {
    render(<DSODataroomView dso={dso} />)

    expect(DataroomHeader).toHaveBeenCalledTimes(1)
    expect(DataroomViewRow).toHaveBeenCalledWith(
      expect.objectContaining({
        title: 'Subscription Document',
        document: dso.subscriptionDocument,
        showDivider: true
      }),
      {}
    )
  })
})
