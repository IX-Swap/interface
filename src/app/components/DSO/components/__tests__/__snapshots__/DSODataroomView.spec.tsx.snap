// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DSODataroomView matches snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
    >
      <h5
        class="MuiTypography-root MuiTypography-h5 css-14eqi03-MuiTypography-root"
      >
        Subscription Document
      </h5>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 css-9llw25-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-10h8c1q-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class=" MuiBox-root css-1r25d6u"
            >
              <div
                class="MuiBox-root css-70qvj9"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-1ncx06u-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    class="css-1s8o7ec"
                    fill="#778194"
                    height="24px"
                    width="24px"
                  >
                    <use
                      href="image-mock#eye"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <p
                  class="MuiTypography-root MuiTypography-body1 label css-rm0xct-MuiTypography-root"
                >
                  File name
                </p>
              </div>
              <div
                class="MuiBox-root css-15aqakd"
              />
              <div
                class="MuiBox-root css-16mkun2"
              >
                <span
                  aria-valuemax="100"
                  aria-valuemin="0"
                  aria-valuenow="0"
                  class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-6gra9t-MuiLinearProgress-root"
                  role="progressbar"
                >
                  <span
                    class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-14iuyd7-MuiLinearProgress-bar1"
                    style="transform: translateX(-100%);"
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
